### PR TITLE
Fix models

### PR DESCRIPTION
**Pull request by dlbrittain**
This is an attempt to reduce code duplication by using metadata defined in custom fields to drive the logic of sqlalchemy model creation from em schemas. 

This should allow for greater flexibility in model creation and reduce the brittleness of the logic.

FYI, This is a work in progress....

TODO: add docstrings and documentation on changes, more tests for edge cases

### CODE REVIEW

File: emannotationschemas/models.py [hunk 1]
```diff
@@ -351,125 +286,109 @@ def create_table_dict(
...
-    return annotation_models.get_model(segmentation_table_name)
+def validate_metadata(model, field, table_metadata):
+    field_metadata = field.metadata.get("metadata")
+    if field_metadata == MetaDataTypes.REFERENCE.value:
+        if type(table_metadata) is not dict:
+            msg = "no metadata provided for reference annotation"
+            raise (InvalidTableMetaDataException(msg))
+        else:
+            try:
+                model["reference_table_name"] = table_metadata["reference_table"]
```
Comment (fcollman) [inline emannotationschemas/models.py:331]: i found this confusing.. model here is a dict?

File: emannotationschemas/models.py [hunk 2]
```diff
@@ -184,22 +162,15 @@ def split_annotation_schema(Schema):
...
-        field_type = type(field)
 
-        # pick off the columns which are the supervoxel and root id of
-        # BoundSpatialPoint fields, and separate them as segmentation columns
-        # TODO: make NumericFields illegal and/or fix logic to pick out
-        # actual fields associated with BoundSpatialPoints
-        if key not in sv_and_root_id_column_keys:
-            annotation_columns[key] = field
-        else:
+        if field.metadata.get("segmentation_field"):
```
Comment (fcollman) [inline emannotationschemas/models.py:170]: what if the field is missing this metadata?

File: emannotationschemas/models.py (hunk 1 above)
Comment (dlbrittain) [reply emannotationschemas/models.py:331]: The arg should be renamed model_dict or something similar...

File: emannotationschemas/models.py (hunk 2 above)
Comment (dlbrittain) [reply emannotationschemas/models.py:170]: Perhaps should leverage the NumericField, or derive a SegmentationField from that... Otherwise I can make more checks, I'm not happy about this filter either...

File: emannotationschemas/models.py [hunk 3]
```diff
@@ -211,102 +185,69 @@ def split_annotation_schema(Schema):
     return flat_annotation_schema, flat_segmentation_schema
 
 
-def create_segmentation_model(
```
Comment (fcollman) [inline emannotationschemas/models.py:214]: this is an example of a function that was used outside that I think you can implement with your new infrastructure, but will minimize the change management if its signature is left unchanged but marked deprecated.
https://github.com/seung-lab/MaterializationEngine/search?q=create_segmentation_model

Comment (fcollman) [reply emannotationschemas/models.py:214]: I'm sorry i see you did that I just wasn't expecting it to move in the file.

Comment (dlbrittain) [reply emannotationschemas/models.py:214]: The materialization methods are mainly wrappers for 'make_annotation_model' and 'make_segmentation_model' since in the mat engine we need to read from the metadata to extract the params to pass to the emschemas methods.
